content/filterx/operator-reference.md

```diff
@@ -0,0 +1,155 @@
+    ```
+
+- For strings (including string values in an object), it concatenates the strings. For example:
+
+    ```shell
+    mystring = "axo";
+    mystring += "flow";
+    # mystring is axoflow
+    ```
+
+- For lists, it appends the new values to the list. For example:
+
+    ```shell
+    mylist = json_array(["one", "two"]);
+    mylist += ["let's", "go"];
+    # mylist is ["one", "two", "let's", "go"]
+    ```
+
+- For a datetime, it increments the time. Note that you can add only integer and double values to a datetime, and:
+
+    - When adding an integer, it must be the number of microseconds you want to add. For example:
+
+        ```shell
+        d = strptime("2000-01-01T00:00:00Z", "%Y-%m-%dT%H:%M:%S%z");
+        d += 3600000000; # 1 hour in microseconds
+        # d is "2000-01-01T01:00:00.000+00:00"
+        ```
+
+    - When adding a double, the integer part must be the number of seconds you want to add. For example:
+
+        ```shell
+        d = strptime("2000-01-01T00:00:00Z", "%Y-%m-%dT%H:%M:%S%z");
+        d += 3600.000; # 3600 seconds, 1 hour
+        # d is "2000-01-01T01:00:00.000+00:00"
+        ```
+
+## Regexp match (equal tilde) {#regexp}
+
+To check if a value contains a string or a regular expression, use the `=~` operator. For example, the following statement is true if the `${MESSAGE}` contains the word `error`:
+
+```shell
+${MESSAGE} =~ "error";
+```
+
+Use the `!~` operator to check if a value doesn't contain an expression. For example, the following statement is true if the `${MESSAGE}` doesn't contain the word `error`:
+
+```shell
+${MESSAGE} !~ "error";
+```
+
+{{% alert title="Note" color="info" %}}
+
+- If you want to process the matches of a search, use the {{% xref "/filterx/function-reference.md#regexp-search" %}} filterx function.
+- If you want to rewrite or modify the matches of a search, use the {{% xref "/filterx/function-reference.md#regexp-subst" %}} filterx function.
+
+{{% /alert %}}
+
+{{< include-headless "chunk/filterx-regexp-notes.md" >}}
+
+<!-- FIXME add some more complex regex examples -->
+
+<!-- 
+FIXME what is relevant/applicable from /chapter-manipulating-messages/regular-expressions/ ?
+
+Is there a workaround for wildcards/globbing? /chapter-routing-filters/filters/regular-expr/_index.md ?
+-->
+
+## Ternary conditional operator
+
+The [ternary conditional operator](https://en.wikipedia.org/wiki/Ternary_conditional_operator) evaluates an expression and returns the first argument if the expression is true, and the second argument if it's false.
+
+Syntax:
+
+```shell
+<expression>?<return-if-true>:<return-if-false>
+```
+
+For example, the following example checks the value of the `${LEVEL_NUM}` macro and returns `low` if it's lower than 5, `high` otherwise.
+
+```shell
+(${LEVEL_NUM} < 5 )?"low":"high";
+```
+
+You can also use it to check if a value is set, and set it to a default value if it doesn't:
+
+```shell
+${HOST} = isset(${HOST})?${HOST}:"default-hostname"
+```
```

Comment:
We can put spaces around the `?`s and `:`s. It might be a bit more readable, IMO.